userspace/libsinsp/parsers.cpp

```diff
@@ -488,6 +488,7 @@ void sinsp_parser::process_event(sinsp_evt *evt)
 		break;
 	case PPME_CONTAINER_JSON_E:
 	case PPME_CONTAINER_JSON_2_E:
+	case PPME_CONTAINER_JSON_2_X:
```

Comment:
What if you simply remove this change? I.e. ignore the event in libsinsp itself, just generate it for whoever wants to consume it? (parse_container_json_evt needs more changes otherwise but I'm not sure we need it at all)

Reply:
Great suggestion! Yeah let's check what's most elegant to still have the payload of the container upon removal when hitting the filterchecks ...

Coming back to our discussion in that issue. Would you accept a change (in a separate PR) that in container parser we DO NOT use add_container and instead first lookup the key in the map and if it exists update the payload (update_container or so) so that adding a new item to the map really just happens in one place ensuring when we remove a container there is absolutely no chance it could possibly be re-added later? Same would apply to user, but that later ... I feel this would be more memory safe? Look at me, looking at this part of the code base first time and am messing up.